tests/tools/clean_orphans_gcp.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import os
+import sys
+
+import google.cloud.compute as compute
+import google.cloud.storage as storage
+import google.oauth2
+
+
+parser = argparse.ArgumentParser(description = 'Cleanup integration test resources in GCP')
+parser.add_argument('-p', '--project', metavar='project_id', required=False, dest='project', default='sap-cp-k8s-gdnlinux-gcp-test',
+                    help='Default value is "sap-cp-k8s-gdnlinux-gcp-test"')
+parser.add_argument('-r', '--region', metavar='region', required=False, dest='region', default='europe-west1', 
+                    help='Default value is "europe-west1"')
+parser.add_argument('-z', '--zone', metavar='zone', required=False, dest='zone', default='europe-west1-d',
+                    help='Default value is "europe-west1-d"')
+parser.add_argument('-a', '--auth', metavar='service_account.json', required=False, dest='auth_path', default='',
+                    help='Path to JSON file that holds the service account login credentials. Defaults to Google application default credentials (ADC).')
+args = parser.parse_args()
+
+if os.path.exists(args.auth_path):
+    credentials = google.oauth2.service_account.Credentials.from_service_account_file(args.auth_path)
+else:
+    credentials = None
+
+try:
+    storage_client = storage.Client(credentials=credentials, project=args.project)
+    image_client = compute.ImagesClient(credentials=credentials)
+    subnet_client = compute.SubnetworksClient(credentials=credentials)
+    firewall_client = compute.FirewallsClient(credentials=credentials)
+    network_client = compute.NetworksClient(credentials=credentials)
+    instance_client = compute.InstancesClient(credentials=credentials)
+except google.auth.exceptions.DefaultCredentialsError as e:
+    print(f"No credentials found. {e}\n\n")
+    parser.print_help()
+    sys.exit(os.EX_CONFIG)
+
+
+def wait_for_operation(operation, name):
+    print(f"Waiting for deletion of resource {name} to complete...")
+    result = operation.result(timeout=60)
```

Comment:
That 60 seconds timeout proved to be a bit too short during my tests - especially when it comes to tearing down VM instances. Maybe we can extend that to 5 Minutes and/or make it configurable...